### PR TITLE
🌱 Bump golangci-lint version (1.48.0 -> 1.49.0)

### PR DESCRIPTION
**Pull request by oscr**
Updates the lint Makefile target updating the golangci-lint version from `1.48.0` to `1.49.0`. 

EDIT: First pr failed due to short description. Hopefully better now. Sorry for retrigger.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -83,7 +83,7 @@ GOLANGCI_LINT = $(shell pwd)/bin/golangci-lint
 golangci-lint:
 	@[ -f $(GOLANGCI_LINT) ] || { \
 	set -e ;\
-	curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(shell dirname $(GOLANGCI_LINT)) v1.48.0 ;\
+	curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(shell dirname $(GOLANGCI_LINT)) v1.49.0 ;\
```
Comment (camilamacedo86) [inline Makefile:86]: We also need to change the github action, @oscr could you please update that as well?

Comment (oscr) [reply Makefile:86]: Thank you @camilamacedo86 for your feedback. Fixed.

**Review by camilamacedo86 — APPROVED** [review]
Thank you for the contribution 🥇 

/lgtm

**Pull request merged**